include/PriorityQueue.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+
+template <class T>
+struct MyComparator {
+	T operator()(T a, T b) const {
+		return a - b;
+	}
+};
+
+
+
+template <class T>
+class PriorityQueue { 
+public:
+	 
+	void push(const T& t); 
+	T poll();
+
+private:
+	std::list<T> m_queue;
+};
+
+template<class T>
+inline void PriorityQueue<T>::push(const T& t)
+{
+	auto iter = m_queue.begin();
+	while (iter != m_queue.end() && MyComparator<T>{}(*iter, t) < 0) {

```

Comment:
You’re initializing the `Comparator` on every iteration which could hurt performance. How could you improve efficiency here?